### PR TITLE
[New Rule] Dynamic Linker Copy

### PR DESCRIPTION
**Pull request by DefSecSentinel**
## Issues

https://github.com/elastic/detection-rules/issues/2081

## Summary

This rule will detect the copying of the Linux dynamic loader binary for the purpose of creating a backup copy. This technique was seen recently being utilized by advanced Linux malware prior to patching the dynamic loader in order to inject and preload a malicious shared object file. This activity should never occur and if it does then it should be considered suspicious or possibly malicious.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Review by imays11 — APPROVED** [review]

File: rules/linux/persistence_dynamic_linker_backup.toml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+language = "eql"
+license = "Elastic License v2"
+name = "Dynamic Linker Copy"
+references = [
+    "https://www.intezer.com/blog/incident-response/orbit-new-undetected-linux-threat/"
+]
+risk_score = 85
+rule_id = "df6f62d9-caab-4b88-affa-044f4395a1e0"
+severity = "high"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Orbit"]
```
Comment (terrancedejesus) [inline rules/linux/persistence_dynamic_linker_backup.toml:22]: I do like the idea of "if we have a specific malware name, CVE or adversary add it to tags". Therefore if I am a customer and I read a report on Orbit, I can review tagged rules with that threat name and enable them or even see if anything triggered recently.

File: rules/linux/persistence_dynamic_linker_backup.toml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type == "start" and process.name : ("cp", "rsync") and process.args : ("/lib/x86_64-linux-gnu/ld-linux-x86-64.so.*")
```
Comment (terrancedejesus) [inline rules/linux/persistence_dynamic_linker_backup.toml:27]: Thoughts on looking for any copy of a shared object from `/lib/x86_64-linux-gnu/` to `/lib/*/*.so`. Looking at Joe's Sandbox report it looks like it copies the shared object to a random character generated folder in `lib/`. Or potentially make it a sequence where process start with `cp` or `rsync` and the legitimate shared object folder, followed but a file creation at `/lib/*/*.so`?

**Review by terrancedejesus — APPROVED** [review]
Added a potential adjustment and question, but this is great!

File: rules/linux/persistence_dynamic_linker_backup.toml (hunk 2 above)
Comment (DefSecSentinel) [inline rules/linux/persistence_dynamic_linker_backup.toml:27]:
```suggestion
sequence by process.entity_id with maxspan=1m
[process where event.type == "start" and process.name : ("cp", "rsync") and process.args : ("/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2", "/etc/ld.so.preload")]
[file where event.action == "creation" and file.extension == "so"]
```
```suggestion
process where event.type == "start" and process.name : ("cp", "rsync") and process.args : ("/lib/x86_64-linux-gnu/ld-linux-x86-64.so.*")
```

File: rules/linux/persistence_dynamic_linker_backup.toml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+[metadata]
+creation_date = "2022/07/12"
+maturity = "production"
+updated_date = "2022/07/12"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects the copying of the Linux dynamic loader binary for the purpose of creating a backup copy. This technique was seen recently being utilized by Linux malware prior to patching the dynamic loader in order to inject and preload a malicious shared object file. This activity should never occur and if it does then it should be considered highly suspicious or malicious.
```
Comment (DefSecSentinel) [inline rules/linux/persistence_dynamic_linker_backup.toml:9]:
```suggestion
Detects the copying of the Linux dynamic loader binary and subsequent file creation for the purpose of creating a backup copy. This technique was seen recently being utilized by Linux malware prior to patching the dynamic loader in order to inject and preload a malicious shared object file. This activity should never occur and if it does then it should be considered highly suspicious or malicious.
```

File: rules/linux/persistence_dynamic_linker_backup.toml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+license = "Elastic License v2"
+name = "Dynamic Linker Copy"
+references = [
+    "https://www.intezer.com/blog/incident-response/orbit-new-undetected-linux-threat/"
+]
+risk_score = 85
+rule_id = "df6f62d9-caab-4b88-affa-044f4395a1e0"
+severity = "high"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Orbit"]
+timestamp_override = "event.ingested"
```
Comment (DefSecSentinel) [inline rules/linux/persistence_dynamic_linker_backup.toml:23]:
```suggestion
```

**Pull request merged**